dependencies.yaml

```diff
@@ -142,17 +142,19 @@ dependencies:
           - --extra-index-url=https://pypi.nvidia.com
           - --extra-index-url=https://pypi.anaconda.org/rapidsai-wheels-nightly/simple
     specific:
+      # very tight >=x.x.x,<x.x.(x+1) here allows for installation of
+      # post release like 1.15.0.post1
       - output_types: [requirements, pyproject]
         matrices:
           - matrix: {cuda: "12.*"}
             packages:
-              - libucx-cu12==1.15.0
+              - libucx-cu12>=1.15.0,<1.15.1
           - matrix: {cuda: "11.*"}
             packages:
-              - libucx-cu11==1.15.0
+              - libucx-cu11>=1.15.0,<1.15.1
           - matrix: null
             packages:
-              - libucx==1.15.0
+              - libucx==1.15.0,<1.15.1
```

Comment:
In this case, should the null matrix entry actually be empty? Since it is valid to build without CUDA, this is a bit different from other RAPIDS libs where we'd specify this matrix entry to satisfy the use case where you're building locally (not wheels) but also building dependencies (e.g. unified devcontainer).

Not a big deal either way since ucx-py is going away soon in favor of ucxx, but it's worth thinking about since this could impact the corresponding ucxx dependencies.